### ISSUE TITLE
Server side redirect return

### ISSUE THREAD

**Issue by saschb2b** [tatethurston/nextjs-routes#56]
We are using redirects for cases when data is not found or not present. Any way here to also type the source and destination parts for the redirect return?

For example

```
export const getServerSideProps: GetServerSideProps<DetailsPageProps> = async (context) => {
  const result = some fetching

  const data = result.data;

  if (!data) {
    return {
      redirect: {
        source: '/data/[id]',
        destination: '/',
        permanent: false,
      },
    };
  }

  return {
    props: {
      data,
    },
  };
};

```

**Comment by tatethurston** [comment]
Hey @saschb2b thanks for opening this. [getServersideProps](https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props#redirect) redirect does not perform the same interpolation as `route` so it can't support dynamic routes as typed by `nextjs-routes`. 

Also question about your example above, where is `source` coming from? getServersideProps' redirect is typed as follows:

```ts
      statusCode: 301 | 302 | 303 | 307 | 308
      destination: string
      basePath?: false
```

**Comment by saschb2b** [comment]
That solution would be superb!

And yes you are right about source. I confused it with static redirects in the nextjs config https://nextjs.org/docs/api-reference/next.config.js/redirects and somehow used it without thinking about it too much. So it's only `destination` that needs to be typed

**Comment by tatethurston** [comment]
This is released in [v0.0.21](https://github.com/tatethurston/nextjs-routes/releases/tag/v0.0.21)
